src/drt/src/RoutingCallBack.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+#include <stdio.h>
+
+#include <boost/asio.hpp>
+#include <mutex>
+
+#include "dr/FlexDR.h"
+#include "dst/Distributed.h"
+#include "dst/JobCallBack.h"
+#include "dst/JobMessage.h"
+#include "triton_route/TritonRoute.h"
+#include "utl/Logger.h"
+namespace odb {
+class dbDatabase;
+}
+namespace fr {
+
+class RoutingCallBack : public dst::JobCallBack
+{
+ public:
+  RoutingCallBack(triton_route::TritonRoute* router,
+                  dst::Distributed* dist,
+                  utl::Logger* logger)
+      : router_(router), dist_(dist), logger_(logger)
+  {
+    globals_path_ = "";
+  }
+  void onRoutingJobReceived(dst::JobMessage& msg,
+                            boost::asio::ip::tcp::socket& sock)
+  {
+    if (msg.type != dst::JobMessage::ROUTING)
+      return;
+    RoutingJobDescription* desc
+        = static_cast<RoutingJobDescription*>(msg.desc.get());
+    if (access(desc->path.c_str(), F_OK) == -1) {
+      logger_->warn(utl::DRT, 605, "Worker file {} not found", desc->path);
+      return;
+    }
+    if (globals_path_ != desc->globals_path) {
+      mx_.lock();
```

Comment:
Can a worker ever be running more than one job at a time?  I'm not sure why the lock is needed.

Reply:
Yes it can be running multiple drWorkers on different threads.